src/site/docs/synonyms/assets/es6-samples.xml

```diff
@@ -0,0 +1,1038 @@
+
+if (!myNaN) {
+  console.log('NaN is treated as false');
+}
+]]></code>
+  </synonym>
+  <synonym id="syn-checking-for-undefined">
+    <title>Checking for undefined</title>
+    <code language="javascript"><![CDATA[
+if (typeof isUndefined === 'undefined') {
+  console.log('undefined');
+}
+]]></code>
+  </synonym>
+  <synonym id="syn-value-and-identity-equality">
+    <title>Value and identity equality</title>
+    <code language="javascript"><![CDATA[
+var letterA = 'A';
+var charA = String.fromCharCode(65);
+
+// JavaScript converts both values to the same type
+// before checking their value with 'double equals'.
+letterA == charA // == true
+
+// Similarly...
+var number5 = 5;
+var char5 = '5';
+
+// This comparison triggers type conversion
+number5 == char5 // == true
+
+// 'triple equals' checks type and value
+letterA === charA // == true
+number5 === char5 // == false
+]]></code>
+  </synonym>
+</theme>
+<theme id="theme-functions">
+  <title>Functions</title>
+  <synonym id="syn-function-definition">
+    <title>Function definition</title>
+    <code language="javascript"><![CDATA[
+function fn() {
+  …
+  return true;
+}
+
+// can also be written as
+function fn() : bool {
+  …
+}
+]]></code>
+  </synonym>
+  <synonym id="syn-return-value">
+    <title>Return value</title>
+    <code language="javascript"><![CDATA[
+function fn() { return 'Hello'; }
+fn(); // == 'Hello'
+
+(function(){})() // == returns undefined
+]]></code>
+  </synonym>
+  <synonym id="syn-assign-a-function-to-a-variable">
+    <title>Assign a function to a variable</title>
+    <code language="javascript"><![CDATA[
+var loudify = (msg) => msg.toUpperCase();
+
+loudify('not gonna take it anymore');
+// NOT GONNA TAKE IT ANYMORE
+]]></code>
+  </synonym>
+  <synonym id="syn-optional-parameters">
+    <title>Optional parameters</title>
+    <code language="javascript"><![CDATA[
+function fn(a, b, c) { return c; };
+
+fn(1) // == undefined
+fn(1, 2, 3) // == 3
+
+function fn(x, ...y) {
+  // y is an Array
+  return x * y.length;
+}
+fn(3, "hello", true) == 6
+
+function fn(x, y, z) {
+  return x + y + z;
+}
+// Pass each elem of array as argument
+fn(...[1,2,3]) == 6
+]]></code>
+  </synonym>
+  <synonym id="syn-default-parameters">
+    <title>Default parameters</title>
+    <code language="javascript"><![CDATA[
+function send(msg, rate = 'First Class') {
+  return `$msg was sent via $rate`;
+}
+
+send('hello') // == 'hello was sent via First Class'
+send("I'm cheap", '4th class') // == "I'm cheap was sent via 4th class
+]]></code>
+  </synonym>
+  <synonym id="syn-named-parameters">
+    <title>Named parameters</title>
+    <code language="javascript"><![CDATA[
+// EcmaScript6 does not have native support for named parameters
+]]></code>
+  </synonym>
+  <synonym id="syn-variable-number-of-arguments">
+    <title>Variable number of arguments</title>
+    <code language="javascript"><![CDATA[
+function superHeros() {
+  for (var i = 0; i < arguments.length; i++) {
```

Comment:
Also, I think `arguments` is discouraged in ES6. The way to write this would be:
```js
function superHeros(...args) {
  for (let arg of args) {
    console.log("There's no stopping " + arg);
  }
}
```